lib/identity_validations/service_provider_validation.rb

```diff
@@ -63,7 +64,8 @@ def saml_client_cert_is_x509_if_present
 
     def uri_valid?(uri)
       parsed_uri = URI.parse(uri)
-      parsed_uri.scheme.present? && parsed_uri.host.present?
+      return false unless parsed_uri.scheme.present?
+      /https?/ =~ parsed_uri.scheme ? parsed_uri.host.present? : true
```

Comment:
another quick thought --- there's the `native` attribute in the IDP database for ServiceProvider, what if we made sure that non-HTTP urls have `native` set to true?

Reply:
That would be fine to add as a new validation, but I don't know that it is consistently applied. We could make a story to add it?

Reply:
ya for sure, if you want to track that as a follow-up that's fine